mergify_engine/clients/github.py

```diff
@@ -0,0 +1,90 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2020 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+from datetime import datetime
+
+import httpx
+
+from mergify_engine import config
+from mergify_engine.clients import common
+from mergify_engine.clients.github_app import GITHUB_APP
+
+
+DEFAULT_ACCEPT = "application/vnd.github.machine-man-preview+json"
+
+
+class GithubInstallationAuth(httpx.Auth):
+    def __init__(self, installation_id):
+        self.installation_id = installation_id
+        self.access_token = None
+
+    def auth_flow(self, request):
+        request.headers["Authorization"] = f"token {self.get_access_token()}"
+        response = yield request
+        if response.status_code == 401:
+            self.access_token = None
+            request.headers["Authorization"] = f"token {self.get_access_token()}"
+            yield request
```

Comment:
I feel like that should be a while loop now?
why only 2 requests otherwise? The last one could also fail no?

Reply:
Well, in theory if a freshly created token still return a 401, that means we not have the permissions anymore.